packages/salesforce-adapter/src/filters/formula_utils/parse.ts

```diff
@@ -0,0 +1,213 @@
+  static STANDARD_FIELD = new IdentifierType('standardFields')
+  static CUSTOM_OBJECT = new IdentifierType('customObjects')
+  static STANDARD_OBJECT = new IdentifierType('standardObjects')
+  static CUSTOM_LABEL = new IdentifierType('customLabels')
+  static CUSTOM_SETTING = new IdentifierType('customSettings')
+  static CUSTOM_METADATA_TYPE_RECORD = new IdentifierType('customMetadataTypeRecords')
+  static CUSTOM_METADATA_TYPE = new IdentifierType('customMetadataTypes')
+  static UNKNOWN_RELATIONSHIP = new IdentifierType('unknownRelationships')
+
+  name: string
+
+  constructor(name: string) {
+    this.name = name
+  }
+}
+
+export type FormulaIdentifierInfo = {
+  type: IdentifierType
+  instance: string
+}
+
+export const parseField = (value: string, object: string | undefined = undefined): FormulaIdentifierInfo => {
+  let actualValue = value
+  if (!value.includes('.')) {
+    if (object === undefined) {
+      log.error('This should not happen!')
+    } else {
+      actualValue = createApiName(object, value)
+    }
+  }
+
+  return {
+    type: (isCustom(actualValue) ? IdentifierType.CUSTOM_FIELD : IdentifierType.STANDARD_FIELD),
+    instance: actualValue,
+  }
+}
+
+export const parseObject = (object: string): FormulaIdentifierInfo => {
+  let type = IdentifierType.STANDARD_OBJECT
+
+  if (isCustom(object)) {
+    type = IdentifierType.CUSTOM_OBJECT
+  } else if (isCustomMetadata(object)) {
+    type = IdentifierType.CUSTOM_METADATA_TYPE
+  } else if (!isStandardRelationship(object)) {
+    type = IdentifierType.UNKNOWN_RELATIONSHIP
+  }
+
+  return {
+    type,
+    instance: object,
+  }
+}
+
+export const parseCustomMetadata = (value: string): FormulaIdentifierInfo[] => {
+  // $CustomMetadata.Trigger_Context_Status__mdt.SRM_Metadata_c.Enable_After_Insert__c
+  const [, sobject, sobjInstance, fieldName] = parts(value)
+
+  return [
+    {
+      instance: createApiName(sobject, sobjInstance),
+      type: IdentifierType.CUSTOM_METADATA_TYPE_RECORD,
+    },
+    {
+      instance: sobject,
+      type: IdentifierType.CUSTOM_METADATA_TYPE,
+    },
+    parseField(fieldName, sobject),
+  ]
+}
+
+export const parseCustomLabel = (value: string): FormulaIdentifierInfo => (
+  {
+    type: IdentifierType.CUSTOM_LABEL,
+    instance: getField(value),
+  }
+)
+
+export const parseCustomSetting = (value: string): FormulaIdentifierInfo[] => {
+  const [, object, field] = parts(value)
+
+  return [
+    {
+      type: IdentifierType.CUSTOM_SETTING,
+      instance: object,
+    },
+    parseField(field, object),
+  ]
+}
+
+export const parseObjectType = (value: string): FormulaIdentifierInfo[] => {
+  // $ObjectType.Center__c.Fields.My_text_field__c
+  const [, sobject, , fieldName] = parts(value)
+
+  return [
+    parseField(fieldName, sobject),
+    parseObject(sobject),
+  ]
+}
+
+export const parseFormulaIdentifier = (variableName: string, originalObject: string): FormulaIdentifierInfo[] => {
```

Comment:
Nit: Can we pass more context here, I think this strings everywhere might be painful in future iterations on this code.
```suggestion
export const parseFormulaIdentifier = (variableName: string, originalObject: ElemId): FormulaIdentifierInfo[] => {
```